### PR TITLE
Provides better React debug stats + React verbose mode

### PR DESCRIPTION
**Pull request by trueadm**
Release notes: Adds React verbose mode

Adds React verbose mode for better debugging.

### CODE REVIEW

**Comment by hermanventer** [comment]
The serializer already has a logging infrastructure.

**Comment by trueadm** [comment]
@hermanventer What logging infrastructure are you referring to specifically. Most of this PR is related to logging React information.

**Comment by gaearon** [comment]
You can search the codebase for `.logger` calls. I think Herman might be asking to reuse the same infra for consistency.

**Comment by trueadm** [comment]
I just wanted to log out some basic information, `Logger` appears to be only able to log out things with given values. I couldn't see any other way of just outputting simple verbose information.

**Comment by hermanventer** [comment]
The Logger API is a bit strange, but passing in a value that is not intrinsic will in effect ignore the value. Mostly, the logger infrastructure separates mechanism from policy (leaving the latter up to the driver program, not the Prepack infrastructure), which is why I'm in favor of using it.

Feel free to make the Logger API a bit less strange.

**Comment by trueadm** [comment]
@hermanventer Gotcha. Let me know if my latest changes make sense then.

File: website/js/repl.js [hunk 1]
```diff
@@ -75,6 +75,12 @@ var optionsConfig = [
     defaultVal: true,
     description: "Removes Flow type annotations from the output."
   },
+  {
+    type: "boolean",
+    name: "abstractEffectsInAdditionalFunctions",
```
Comment (hermanventer) [inline website/js/repl.js]: Additional -> Pure

Comment (trueadm) [reply website/js/repl.js]: This is the name of the option in `options.js`.

**Pull request closed without merging**